snuba/datasets/schemas/__init__.py

```diff
@@ -66,6 +67,18 @@ def supports_sample(self) -> bool:
         """
         return True
 
+    @abstractmethod
+    def get_promoted_columns_spec(self) -> Mapping[str, PromotedColumnSpec]:
```

Comment:
Not all RelationalSource are supposed to have this, the alternative would have been to subclass every RelationalSource (there are 4). And expose the same alternative to the TableSchema (and its subclasses). I don't think it is worth doing that, it actually makes the hierarchy worse.